runtime/JavaScript/src/antlr4/CharStreams.d.ts

```diff
@@ -1,4 +1,9 @@
-import {CharStream} from "./CharStream";
+/* Copyright (c) 2012-2022 The ANTLR Project. All rights reserved.
+ * Use of this file is governed by the BSD 3-clause license that
+ * can be found in the LICENSE.txt file in the project root.
+ */
+
+import CharStream from "./CharStream.js";
```

Comment:
We're not moving away from typescript style imports. Not sure what problem this is trying to solve but in my experience it's a recipe for problems.

Reply:
What is a "typescript import"? This is a normal JS default import as being used in many other places in the JS runtime.

Reply:
this file is a ts file, not a js file.
As such it needs to follow ts import conventions i.e. no suffix.
It makes no sense to import a js file in a d.ts file in the first place, since we're only looking to import types, not implementations. Think of the d.ts files as a set of files that need to live somewhat independently of the js ones, because the js ones get babelized, uglified and ultimately packaged - they might not exist in the distribution

Reply:
This is irrelevant here, but FYI for normal TS files, the extension is also not necessary and should be avoided, unless the imported file is actually a JS one. See for example https://github.com/hyperledger/aries-framework-javascript/blob/main/packages/core/src/agent/Agent.ts.

Reply:
A link to a random framework doesn't help here. I thought I had provided the link, but cannot find it here. So here it is again. Full ESM support [requires the extension](https://nodejs.org/api/esm.html#mandatory-file-extensions) in Node.js (in the browser they are required anyway). It's the bundler that automatically handles that, but when running with tools like ts-node or tsx they are required.

Hence I'm not going to remove the extensions again.

Reply:
The link you provide is for node i.e. javascript, not for typescript.
This is [the official reference](https://www.typescriptlang.org/docs/handbook/modules.html#import) and you'll read that they don't use extensions except for side effects.

Reply:
But ultimately all TS files end up as JS files, so this applies anyway. If you are against including the .js extension, why did you include them in the existing JS/TS runtime? You even use the extension in the test cases (see Test.ts.stg).

The main point here is: when using tools like ts-node to run TS directly you need the extensions or they will not find the imports. I've been through that for a number of times already.

We are wasting our time here on something which is an absolute minor aspect and I have good reasons to keep the extensions included.

Reply:
@mike-lischke The docs page you link is a little confusing, it makes it seem like the it uses a ".js" extension for side effects only but that is not the case. 

In fact the typescript documentation has a section on file extensions in imports [here](https://www.typescriptlang.org/docs/handbook/esm-node.html), where they explain why the `.js` extension is needed to support both commonjs and esm for node when using typescript.